### PR TITLE
Check null stops

### PR DESCRIPTION
**Pull request by kaelgreco**
Caught a bug in the gtfs link mapper for Sacramento where a trip did not have any associated stops. Seems to be fine if we ignore bad trips in the csv output

### CODE REVIEW

File: web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java [hunk 1]
```diff
@@ -224,33 +224,35 @@ public void setGtfsLinkMappings() {
...
-
-                    Stop stop = stopStopPair.getLeft();
-                    Stop nextStop = stopStopPair.getRight();
-
-                    // format: "{feed_id}:{route_id}/{feed_id}:{stop_id}/{feed_id}:{next_stop_id}"
-                    String transitEdgeString = stop.feed_id + ":" + routeId + "/" + stop.feed_id + ":"
-                            + stop.stop_id + "/" + stop.feed_id + ":" + nextStop.stop_id;
-
-                    rowsForFeed.add(getCsvLine(routeId, stop.feed_id, stop.stop_id, nextStop.stop_id,
+	    if (routeIdToTripsInRoute.containsKey(routeId)) {
```
Comment (marcopalmeri) [inline web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java:227]: nit:
`
if (!routeIdToTripsInRoute.containsKey(routeId)) {
    continue
}
`
if you want one less indent in the code.

**Review by marcopalmeri — APPROVED** [review]

**Comment by kaelgreco** [comment]
@mark-idleman my local idls is out of sync and it looks like the original commit that it's tied too (8e5403bcf1107114a893f43b3fc3f23a96ed7623) "does not belong to any branch on this repository, and may belong to a fork outside of the repository." Is it okay to use master? if not we should just update master to get them in sync

File: web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java [hunk 2]
```diff
@@ -224,7 +224,10 @@ public void setGtfsLinkMappings() {
                                                    HTreeMap<String, String> gtfsLinkMappings) {
         List<String> rowsForFeed = Lists.newArrayList();
         for (String routeId : routeIdToTripsInRoute.keySet()) {
-            for (String tripIdInRoute : routeIdToTripsInRoute.get(routeId)) {
+	    if (!routeIdToTripsInRoute.containsKey(routeId)) {
```
Comment (mark-idleman) [inline web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java:227]: How is this even possible? `routeId` comes from the key set of `routeIdToTripsInRoute`, so `routeIdToTripsInRoute.containsKey(routeId)` has to be true, right?

**Review by mark-idleman — CHANGES_REQUESTED** [review]

**Comment by mark-idleman** [comment]
Hmm, not sure about your idls issue @kaelgreco  - it looks like both graphhopper and model repo master are on https://github.com/replicahq/idls/tree/8e5403bcf1107114a893f43b3fc3f23a96ed7623, the commit you list. Maybe it's just a weird git issue w/ your local idls subdirectory or something?

File: web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java (hunk 2 above)
Comment (mark-idleman) [reply web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java:227]: From [here](https://replicahq.slack.com/archives/CK358RL0Z/p1614849937013400?thread_ts=1600553031.022500&cid=CK358RL0Z), it looks like the original error appeared on this line: 

```for (Pair<Stop, Stop> stopStopPair : tripIdToStopPairsInTrip.get(tripIdInRoute)) {```

So, I think we want to do a check if `tripIdToStopPairsInTrip.containsKey(tripIdInRoute)`

Comment (mark-idleman) [reply web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java:227]: For some additional context:

The origin of `routeIdToTripsInRoute` comes from taking every route ID from `routes.txt`, and mapping them to trips that appear in `trips.txt` and have a `route_id` that lines up with the ID from `routes.txt`. 

`tripIdToStopPairsInTrip` comes from looking in `stop_times.txt`, making a mapping where the keys are the trip IDs appearing in the `trip_id` column for each "stop time" and the values are the set of "stop times" that have that trip ID, and then ordering each list of stop times (ie, the values in this intermediate map we've made) by departure time and grabbing their lat/lons to get a final mapping of (trip ID) --> (set of sequentially-ordered stop-stop pairs) (this is what `tripIdToStopPairsInTrip` holds). We use those sequentially-ordered stop-stop pairs to then do the actual routing between each set of stops (for each trip) later on.

So, I think the cause of the bug must be that there's a trip in trips.txt that doesn't appear in the `trip_id` column of any stop time in `stop_times.txt`. Not well-formatted gtfs, but `continue` definitely seems like the right behavior here.

Comment (kaelgreco) [reply web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java]: You’re right mark. When i was originally debugging the err i fixed/tested this on a different branch that ultimately had too many merge conflicts. i attempted to remake the change from memory but put it on the wrong map 😅  Just tested again now.

Can’t wait for those tests ;)

**Review by mark-idleman — APPROVED** [review]
I think indents got off a bit so just pushed a fix, but otherwise this LGTM! :D

**Pull request merged**